### ISSUE TITLE
build passes in spite of failing tests

### ISSUE THREAD

**Issue by fa-me** [Sh4pe/pynance#41]
Failing tests still allow the build to pass.
Output from Travis:

```
$ doit test
.  test:unit
.........................EE.
======================================================================
ERROR: test_dkb_date_formatting (pynance.parser_test.ParserTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/Sh4pe/pynance/pynance/parser_test.py", line 40, in test_dkb_date_formatting
    def test_dkb_date_formatting(self, dt):
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/hypothesis/core.py", line 1002, in wrapped_test
    raise the_error_hypothesis_found
  File "/home/travis/build/Sh4pe/pynance/pynance/parser_test.py", line 47, in test_dkb_date_formatting
    date_result = formatter(datestr)
  File "/home/travis/build/Sh4pe/pynance/pynance/textimporter.py", line 151, in to_datetime64
    date = datetime.datetime.strptime(datestring, "%d.%m.%Y")
  File "/opt/python/3.6.3/lib/python3.6/_strptime.py", line 565, in _strptime_datetime
    tt, fraction = _strptime(data_string, format)
  File "/opt/python/3.6.3/lib/python3.6/_strptime.py", line 362, in _strptime
    (data_string, format))
ValueError: time data '01.01.999' does not match format '%d.%m.%Y'
======================================================================
ERROR: test_dkb_float_formatting1 (pynance.parser_test.ParserTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/Sh4pe/pynance/pynance/parser_test.py", line 22, in test_dkb_float_formatting1
    allow_nan=False,
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/hypothesis/core.py", line 1002, in wrapped_test
    raise the_error_hypothesis_found
  File "/home/travis/build/Sh4pe/pynance/pynance/parser_test.py", line 29, in test_dkb_float_formatting1
    locale.setlocale(locale.LC_ALL, 'de_DE')
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/locale.py", line 598, in setlocale
    return _setlocale(category, locale)
locale.Error: unsupported locale setting
----------------------------------------------------------------------
Ran 28 tests in 1.393s
FAILED (errors=2)
.  test:integration
### Integration tests are not yet implemented ###
The command "doit test" exited with 0.
after_success
35.16s$ codecov
      _____          _
     / ____|        | |
    | |     ___   __| | ___  ___ _____   __
    | |    / _ \ / _  |/ _ \/ __/ _ \ \ / /
    | |___| (_) | (_| |  __/ (_| (_) \ V /
     \_____\___/ \____|\___|\___\___/ \_/
                                    v2.0.15
==> Detecting CI provider
    Travis Detected
    Fixing merge commit SHA
==> Preparing upload
==> Processing gcov (disable by -X gcov)
    Executing gcov (find /home/travis/build/Sh4pe/pynance -not -path './bower_components/**' -not -path './node_modules/**' -not -path './vendor/**' -type f -name '*.gcno'  -exec gcov -pb  {} +)
==> Collecting reports
    Generating coverage xml reports for Python
    + /home/travis/build/Sh4pe/pynance/coverage.xml bytes=8109891
==> Appending environment variables
    + TRAVIS_OS_NAME
    + TRAVIS_PYTHON_VERSION
==> Uploading
    .url https://codecov.io
    .query commit=96aa31eef44e6fb926acf299074a426cd33af3d7&branch=master&job=491268789&pr=40&service=travis&build=194.1&slug=Sh4pe%2Fpynance&package=py2.0.15
    Pinging Codecov...
    Uploading to S3...
    https://codecov.io/github/Sh4pe/pynance/commit/96aa31eef44e6fb926acf299074a426cd33af3d7
Done. Your build exited with 0.
```

**Comment by Sh4pe** [comment]
I've noticed the following behaviour: 

* If a testcase fails (e.g. with `self.fail()` in one of the unit test methods), `doit test` will return with != 0
* If a testcase fails with an exception (e.g. by throwing one manually), `doit test` returns with 0.

**Comment by Sh4pe** [comment]
See also [this](https://stackoverflow.com/questions/24972098/unit-test-script-returns-exit-code-0-even-if-tests-fail) Stackoverflow thread. Thanks @fa-me!

**Comment by fa-me** [comment]
what do you think of using `pytest` instead of the `unittests.py`. I think it will make everything a bit cleaner - and it's really widely used in conjunction with travis (see for example https://github.com/jakevdp/travis-python-template)
`pytest` usually also gives nice IDE integration (at least with vscode)

**Comment by Sh4pe** [comment]
> what do you think of using `pytest` instead of the `unittests.py`. I think it will make everything a bit cleaner - and it's really widely used in conjunction with travis (see for example https://github.com/jakevdp/travis-python-template)
> `pytest` usually also gives nice IDE integration (at least with vscode)

I'm open to using `pytest`. Maybe one of us should make a PR later and link it here?

**Comment by fa-me** [comment]
started a branch for this

**Comment by fa-me** [comment]
fixed in #46
